### PR TITLE
Display group history

### PR DESCRIPTION
**Pull request by paradime**
image

### CODE REVIEW

**Comment by kwm4385** [comment]
Nice. Maybe some color coding and humanizing the actions would make it more readable.

**Comment by paradime** [comment]
image

File: src/routes/Group/components/TicketView.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      <tr>
+      <td>{action.actionTaken}</td>
+      <td>{action.timestamp}</td>
+      <td>{action.userId}</td>
+      </tr>
+    )
+  }
+
+  userLink(user) {
+    return user ? <a href={'/user/' + user._id +'/'}>{user.name}</a> : <span>No one</span>;
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: Instead of normal anchors which reload the entire page, you should use `Link` components which take advantage of the router. See https://github.com/PagerNation/Escalator-UI/blob/master/src/routes/Home/components/GroupsInfo.js#L14

File: src/routes/Group/components/TicketView.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  ticketRow(ticket, index){
+    var title = (typeof ticket.metadata.title != undefined) ? ticket.metadata.title : 'No name';
+    var rows = [];
+    if (index == 0) {
+      rows.push(<Table.Cell rowSpan={ticket.actions.length}>{title}</Table.Cell>)
+    }
+    if (typeof ticket.actions != 'undefined'){
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: you can shorten this to just `if(ticket.actions)`

File: src/store/ticket.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export const FETCH_TICKETS_SUCCESS = 'FETCH_TICKETS_SUCCESS';
+
+// ------------------------------------
+// Actions
+// ------------------------------------
+export const fetchGroupTickets = (groupName) => {
+  return (dispatch, getState) => {
+    return new Promise((resolve) => {
+      getJSON(`ticket/all?groupName=${groupName}`).then((response) =>{
+        console.log(response[0])
```
Comment (kwm4385) [inline src/store/ticket.js]: don't forget to remove these logs

**Comment by paradime** [comment]
image

**Comment by jarrydlee** [comment]
Are the borders around the empty device types lighter than the rest?

**Comment by paradime** [comment]
image
Good call, it actually wasn't adding a cell at all

**Comment by paradime** [comment]
@kwm4385 How do you feel about this bad boy?

File: src/routes/Group/components/TicketView.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  render() {
+    var rows = []
+    for(var ticket of this.props.tickets) {
+      for(var index in ticket.actions) {
+        rows.push(this.ticketRow(ticket, index))
+      }
+    }
+    return <div>
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: jsx returns should be enclosed in ()

File: src/routes/Group/components/TicketView.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  };
+
+  ticketRow(ticket, index){
+    var title = (ticket.metadata.title) ? ticket.metadata.title : 'No name';
+    var rows = [];
+    var id = 'ticket_'+ticket._id+'_'+index;
+    if (index == 0) {
+      rows.push(<Table.Cell key={'title_'+id} rowSpan={ticket.actions.length}>{title}</Table.Cell>)
+    }
+    if (ticket.actions){
```
Comment (kwm4385) [inline src/routes/Group/components/TicketView.js]: need a space before the {. Make sure to lint files for things like this in general.

File: src/routes/Group/containers/GroupContainer.js [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
   updateEscalationPolicy,
+  fetchGroupTickets,
   processRequest
 };
 
 const mapStateToProps = (state) => ({
   group: state.group,
-  user: state.user
+  user: state.user,
+  tickets: state.ticket,
```
Comment (kwm4385) [inline src/routes/Group/containers/GroupContainer.js]: extra comma
